src/components/WrapperApiReference.js

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { Article } from "basics/Text";
+import { CustomColumn } from "components/ApiReference/SharedStyles";
+
+const RIGHT_COLUMN_COMPONENTS_NAME = {
+  CodeExample: "CodeExample",
+  EndpointsTable: "EndpointsTable",
+  ExampleResponse: "ExampleResponse",
+  MethodTable: "MethodTable",
+};
+// API reference has 2 columns, with certain types of content always being in one
+// or the other. This component sorts through its children to separate them.
+export const WrapperApiReference = ({ children, ...props }) => {
+  const rightColumnContent = React.useMemo(
+    () =>
+      React.Children.toArray(children).filter(
+        (child) => RIGHT_COLUMN_COMPONENTS_NAME[child.props.mdxType],
+      ),
+    [children],
+  );
+  const MiddleColumnContent = React.useMemo(
+    () =>
+      React.Children.toArray(children).filter(
+        (child) => !RIGHT_COLUMN_COMPONENTS_NAME[child.props.mdxType],
+      ),
+    [children],
+  );
```

Comment:
we're just splitting into 2 arrays right? 
might be a little cleaner to use 

_.partition https://lodash.com/docs/4.17.15#partition

or just plain JS to divide: https://stackoverflow.com/a/47225591

Reply:
Oh good call, partition is def the right tool